src/main/java/com/slack/eithernet/ApiResult.kt

```diff
@@ -151,6 +162,7 @@ public sealed interface ApiResult<out T : Any, out E : Any> {
      */
     public class ApiFailure<E : Any> internal constructor(
       public val error: E?,
+      override val exception: ApiException,
```

Comment:
This isn't something we want to expose here. The convenience sought here can just be done via utility extension function, so I don't want to add this extra API surface area here and have an awkward case like this. Especially as this leads to creating ad-hoc exceptions in some of the factory creators below